index.js

```diff
@@ -124,7 +142,51 @@ function maybeSignWebBundle(webBundle, opts, infoLogger) {
   return signedWebBundle;
 }
 
-function validateIWAOptions(opts) {
+function checkIwaOverrideHeaders(headers) {
+  for (const iwaHeaderName of Object.keys(invariableIwaHeaders)) {
+    if (headers[iwaHeaderName] !== invariableIwaHeaders[iwaHeaderName]) {
+      throw new Error(
+        `For Isolated Web Apps ${iwaHeaderName} should be ${invariableIwaHeaders[iwaHeaderName]}. Now it is ${headers[iwaHeaderName]}. If you are bundling a non-IWA, set integrityBlockSign { isIwa: false } in your plugins configs.`
+      );
+    }
+  }
+
+  // TODO: Parse and check `Content-Security-Policy` value.
+  const cspHeaderName = Object.keys(csp)[0];
+  if (!headers[cspHeaderName]) {
+    throw new Error(
+      `For Isolated Web Apps ${cspHeaderName} should not be empty. Default value can be used: ${JSON.stringify(
+        csp[cspHeaderName]
+      )}. In case you are bundling a non-IWA, set integrityBlockSign { isIwa: false } in your plugins configs.`
+    );
+  }
+}
+
+function maybeSetIwaDefaults(opts) {
+  if (
+    opts.integrityBlockSign.isIwa !== undefined &&
+    !opts.integrityBlockSign.isIwa
+  ) {
+    return;
+  }
+
+  // `isIwa` is defaulting to `true` if not provided as currently there is no
+  // other use case for integrityBlockSign outside of IWAs.
+  opts.integrityBlockSign.isIwa = true;
+
+  if (opts.headerOverride === undefined) {
+    console.info(
+      `Setting the empty headerOverrides to IWA defaults. To bundle a non-IWA, set \`integrityBlockSign { isIwa: false }\` in your plugin configs. Defaults are set to:\n ${JSON.stringify(
+        iwaHeaderDefaults
+      )}`
+    );
+    opts.headerOverride = Object.assign({}, iwaHeaderDefaults);
```

Comment:
Silly question, but why the `Object.assign`? If you are worried about it being modified, then maybe `Object.freeze()` would be more appropriate? (Also, `Object.assign` does not prevent someone from modifying the `iwaHeaderDefaults.csp`)